samples/data-sealing/enc1/ecalls.cpp

```diff
@@ -1,10 +1,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#include <common/datasealing_t.h>
```

Comment:
Yes, the discrepancy is caused by using oeedgr8r correctly in the CMake sample: the headers `datasealing_t.h` and `datasealing_u.h` are generated in the associated _build_ directory for `libcommon`, whereas the GNU Make build generates them in place (polluting the source tree). So with `libcommon` being a separate library, it made more sense (to me) to turn these into `<common/*>` headers, where `libcommon` has the correct pair of folders in its `INTERFACE_INCLUDE_DIRECTORIES`. This means we don't have to specify a build directory (which could be _anywhere_), and the headers are grouped by their library `libcommon`, and the _only_ change necessary to the GNU Makefiles is the use of `-I..`

In short, it was the simplest solution I came up with. If you want to provide another one, I'm all ears 😄 but it is more complicated than it looks!

Reply:
I'm more concerned that the nuances of this are documented somewhere for maintainers so we can scale this to more sample authors. If you could capture this somewhere for reference, it would be helpful.

Reply:
Most of this can be avoided with better GNU Make builds for the samples, which @EMumau was going to take a look at. I'll file an issue.

Reply:
#1562